project-koku/koku
# [COST-4604] - Catch more failures correctly for observability

## Jira Ticket

[COST-4604](https://issues.redhat.com/browse/COST-4604)

## Description

This change will add a few more places to correctly update report statuses when processing fails 

## Testing

Scenario 1 (failed to upload files)
1. Checkout Branch
2. Restart Koku
3. Stop minio and attempt to ingest some data
4. We should raise an exception for failed to upload and update the report status correctly
5. Check report status has failed_status set

Scenario 2 (CSV with incompatible columns)
1. Checkout branch
2. Restart koku
3. Create AWS provider
4. Generate some nise data and manually change the values in each column to incorrect data (Set everything to the same string)
6. Trigger download
7. We should fail to process the data but actually capture the problem and correctly update the report status
8. Check report status has failed_status set

## Notes
If you want a super easy way to test No.2 Give me a shout, we can use the customer filter flow and I have a payload in a bucket already.

...
